contents/product-engineers/llm-product-metrics.md

```diff
@@ -0,0 +1,157 @@
+---
+date: 2024-05-03
+title: 'Product metrics to track for LLM apps'
+author:
+  - lior-neu-ner
+featuredImage: >-
+  https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/posthog-marketing/marketing-hog.jpg
+featuredImageType: full
+category: Engineering
+tags:
+  - Guides
+  - Product metrics
+crosspost:
+  - Blog
+---
+
+import { ProductScreenshot } from 'components/ProductScreenshot'
+
+LLM-powered apps differ in three key ways when compared to regular SaaS apps:
+
+1. Their costs increase as as app usage increases.
+2. Request latency spikes and errors are more common.
+3. You're not always sure if your LLM output is what was expected.
+
+This means that in addition to [regular product metrics](https://posthog.com/product-engineers/product-health-metrics), you need to monitor LLM-specific ones too. To understand which ones to track, we've put together a list of useful metrics to track.
+
+The metrics are grouped into three categories: [cost](#cost-related-metrics), [usage](#usage-metrics), and [debugging](#debug-metrics).
+
+## Cost-related metrics
+
+### Average cost per user
+
+**What is it?** Your total LLM costs divided by the number of active users.
+
+**Why it's useful:** Gives you an idea of how your costs will grow with your product. You can also compare this to revenue per user to understand if your profit margin is viable.
+
+**Questions to ask:**
+- Are average costs going up or down over time? 
+- If they're increasing over time, is it because of increased product usage or increased generation costs?
+- What specific features or types of interactions contribute the most to the cost per user?
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/v1714658975/posthog.com/contents/average-cost-per-user-light.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/v1714658975/posthog.com/contents/average-cost-per-user-dark.png"
+  alt="Line chart showing average LLM cost per user over time"
+/>
+
+### P95 cost per user
+
+**What is it?** The top 5% of users who are consuming the most LLM costs.
+
+**Why it's useful:** Determines if specific users disproportionately affect overall costs. You can then investigate whether these costs are justified by their usage, or if there are inefficiencies to address.
```

Comment:
Combine with session replays?

Reply:
hmm tried to tie them in but it comes across as too marketing-y